### PR TITLE
test(fuzz): add FuzzScraperConfigRoundTrip for ScraperConfig decode + validate

### LINKED ISSUE #1367: test(fuzz): add FuzzScraperConfigRoundTrip for scraper.UnmarshalScraperConfig + ValidateConfig
## Goal
Add a fuzz test for the scraper config decoder + validator. `ScraperConfig` is round-tripped through the DB (JSON column) and user edits land here via the API. Bot reviewers have repeatedly flagged enum drift on `FieldName` / `ProviderName` here.

## Target
- `internal/scraper/service.go:131,153` (`UnmarshalScraperConfig`)
- `internal/scraper/config.go:54` (`ValidateConfig`)

## Suggested seed corpus
- `DefaultConfig()` serialized.
- Configs with unknown provider names.
- Configs with empty fallback chains.
- Configs with provider names containing JSON metacharacters.
- Configs with duplicate `FieldConfig` entries.
- Configs with `fallback_chains` slice containing nil providers.
- Very large field arrays.

## Acceptance criteria
- [ ] New `FuzzScraperConfigRoundTrip` in `internal/scraper/`.
- [ ] Asserts that `Validate(Unmarshal(input))` either returns a typed error or produces a struct that round-trips through `Marshal`.

## Charter
Direct M49 fit -- closes the enum/discriminator-drift surface bot reviewers keep flagging. Plan issue A7.

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary
- M49 W3 sub-task 3F. Adds `FuzzScraperConfigRoundTrip` exercising `json.Unmarshal` into `ScraperConfig` plus `ValidateConfig` (config.go:54), with a round-trip assertion: when both succeed, the config must `Marshal -> Unmarshal` to an equivalent struct.
- 12 seeds covering `DefaultConfig()`, unknown provider names, empty fallback chains, JSON metacharacters in provider names, duplicate FieldConfig entries, nil providers in fallback chains, very large field arrays, empty JSON `{}`, empty bytes, and `null` fallback_chains.
- ~122,000 execs/10s on local smoke run.

## Note on target
Issue #1367 cited `service.go:131,153` as `UnmarshalScraperConfig`, but that named function doesn't exist in the codebase -- the actual call sites are inline `json.Unmarshal` inside `loadConfig` (line 137) and `loadConfigWithOverrides` (line 159). The fuzz target exercises the same code path (`json.Unmarshal` into `ScraperConfig` then `ValidateConfig`), giving equivalent coverage of the enum / discriminator drift surface that bot reviewers have flagged.

## Round-trip assertion
Per-field equality. `ID`, `Scope`, `Fields[i].{Field,Primary,Enabled,Category}`, and `FallbackChains[i].{Category,Providers[j]}` compared with `==`. `CreatedAt` / `UpdatedAt` use `.Equal()` rather than `==` to avoid spurious failures from JSON's `time.Location` pointer differences after decode.

## Test plan
- [x] `go test ./internal/scraper/...`
- [x] `go test -fuzz=^FuzzScraperConfigRoundTrip$ -fuzztime=10s ./internal/scraper` -> 122k execs, no panics
- [x] `golangci-lint run ./internal/scraper/...` clean

Closes #1367.

## Summary by CodeRabbit

* **Tests**
  * Added a fuzz test for scraper configuration round-trip serialization. It seeds diverse JSON inputs (including malformed and edge cases), validates successful configs, and asserts semantic equivalence after serialize/deserialize — checking IDs, timestamps, fields, and fallback chains to improve robustness and data integrity.

[Review Change Stack](https://app.coderabbit.ai/change-stack/sydlexius/stillwater/pull/1475)

### CODE REVIEW

File: internal/scraper/config_fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+			// Invalid JSON: acceptable, no panic is the invariant here.
+			return
+		}
+
+		// Step 2: validate -- typed errors are acceptable; panics are not.
+		if err := ValidateConfig(&cfg1); err != nil {
+			// Typed validation error: the fuzz engine has found an invalid but
+			// parseable config. This is expected and correct behavior.
+			return
+		}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/scraper/config_fuzz_test.go:86]: `ValidateConfig` (config.go:54-78) returns only bare `fmt.Errorf` — the package defines no typed validation error, so `errors.As` has no target to assert against. Adding one would be architectural and out-of-scope for a fuzz-test PR. `ValidateConfig` is a pure in-memory enum check with no I/O paths, so there is no "unexpected internal failure" mode distinct from validation failure. The current early-return prevents structurally invalid inputs from polluting the fuzz corpus while still ensuring upstream decoders cannot panic on them, which is the contract the fuzz target enforces.
